app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt

```diff
@@ -279,3 +301,41 @@ enum class UrlTextInputLocation(internal val extra: String) {
     HOME("home"),
     MENU("menu"),
 }
+
+/** Counts the number of unique home tiles per session the user has clicked on. */
+@UiThread // We get-and-set over SharedPreferences in countTile so we need resource protection.
+private object TelemetryHomeTileUniqueClickPerSessionCounter {
+
+    private const val COLLECTION_DISABLED = -1
+
+    fun countTile(context: Context, tile: HomeTile) {
+        Assert.isUiThread()
+        if (!TelemetryHolder.get().configuration.isCollectionEnabled) { return }
+
+        val sharedPrefs = getSharedPrefs(context)
+        val clickedTileIDs = (sharedPrefs.getStringSet(KEY_CLICKED_HOME_TILE_IDS_PER_SESSION, null)
+                ?: setOf()).toMutableSet() // create a copy: we can't modify a SharedPref StringSet.
+        clickedTileIDs.add(tile.idToString())
```

Comment:
To be a little more efficient, you could add this only if it's not already in the set, so we don't have to re-write the set.